tsdb/querier.go

```diff
@@ -326,50 +326,41 @@ func findSetMatches(pattern string) []string {
 // PostingsForMatchers assembles a single postings iterator against the index reader
 // based on the given matchers.
 func PostingsForMatchers(ix IndexReader, ms ...labels.Matcher) (index.Postings, error) {
-	var its, notIts []index.Postings
-	// See which label must be non-empty.
-	labelMustBeSet := make(map[string]bool, len(ms))
```

Comment:
Removing this removes an optimisation. We want to be smart for things like {a=~".",a!="1"}

Reply:
Thanks for reminding me. Finally I know why the code design like this 😂 

If we could specify same label for multiple times, then everything make sense.

But if query like `{a="1", a="2"}` reach here, then the result will be incorrect, so PromQL will filter query like this ?

Hmm.. then this PR seems unnecessary, maybe could add some comment to make the code more clear 😄

Reply:
The label must be set in that case, though it'll match nothing.

Reply:
Yes, forget the final intersect :)